chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt

```diff
@@ -758,30 +762,120 @@ class ChatroomDetailFragment :
     private fun initAttachmentClick() {
         binding.apply {
             inputBox.ivAttachment.setOnClickListener {
-                if (!viewModel.isAudioSupportEnabled()) {
-                    layoutAttachments.ivAudio.hide()
-                    layoutAttachments.tvAudioTitle.hide()
-                }
                 initVisibilityOfAttachmentsBar(View.VISIBLE)
             }
         }
     }
 
-    private fun initAttachmentsView() {
-        binding.layoutAttachments.apply {
-            ivGallery.setOnClickListener {
+    /**
+     * create the list of supported attachments as per the chatroom
+     * Case 1 -> If it is Group Chatroom
+     * Allow all the attachments, but if poll and audio is enabled from backend
+     *
+     * Case 2 -> If it is 1-1 Chatroom with user
+     * Allow all the attachments, except Polls
+     *
+     * Case 3 -> If it is 1-1 Chatroom with AI Chatbot
+     * Allow only Camera, Gallery (images only) and Audio
+     *
+     * Case 4 -> If Custom widget is enabled from backend
+     * Add custom widget allow.
+     */
+    private fun getSupportedAttachmentTypes(): List<LMChatAttachmentPickerItemViewData> {
+        val supportedAttachments = mutableListOf<LMChatAttachmentPickerItemViewData>()
+
+        //add camera
+        val cameraAttachment = LMChatAttachmentPickerItemViewData.Builder()
+            .attachmentType(LMChatAttachmentType.CAMERA)
+            .attachmentIcon(R.drawable.lm_chat_ic_create_camera)
+            .attachmentName(requireContext().getString(R.string.lm_chat_camera))
+            .build()
+
+        //add gallery
+        val galleryAttachment = LMChatAttachmentPickerItemViewData.Builder()
+            .attachmentType(LMChatAttachmentType.GALLERY)
+            .attachmentIcon(R.drawable.lm_chat_ic_create_gallery)
+            .attachmentName(requireContext().getString(R.string.lm_chat_gallery))
+            .build()
+
+        //add document
+        val documentAttachment = LMChatAttachmentPickerItemViewData.Builder()
+            .attachmentType(LMChatAttachmentType.DOCUMENT)
+            .attachmentIcon(R.drawable.lm_chat_ic_document_new)
+            .attachmentName(requireContext().getString(R.string.lm_chat_document))
+            .build()
+
+        //add audio
+        val audioAttachment = LMChatAttachmentPickerItemViewData.Builder()
+            .attachmentType(LMChatAttachmentType.AUDIO)
+            .attachmentIcon(R.drawable.lm_chat_ic_create_music)
+            .attachmentName(requireContext().getString(R.string.lm_chat_audio))
```

Comment:
move the initialisation inside their respective if checks. No sense of creating objects and not using them if the attachment type is not supported.